### PR TITLE
Budget app build

### PR DESCRIPTION
**Pull request by eudondian**
# In This Project:
#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

https://www.loom.com/share/252d74700f0e4d8483921786fc145f75?sid=ef4b250a-e7cb-408a-84b7-f0a05d320657

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <%= javascript_importmap_tags %>
+  </head>
+
+  <body>
+  <p class="notice"><%= notice %></p>
+  <p class="alert"><%= alert %></p>
+
+   
+    <%= yield %>
+  </body>
```
Comment (anagudelogu) [inline app/views/layouts/application.html.erb]: - [ ] Please consider fixing your UI. All of your views have a blank space at the top of the screen:

image 

Remember that is a requirement following as closely as possible the given design guidelines. Please fix it! 🤓 🔧

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          <h2 class="ms-5 md-3"><%= category.name %></h2>
+        </div>
+        <p class="mb-0">
+          $ <%= category.expenses.sum(:amount) %>
+        </p>
+      </div>
+      <small><%= category.created_at.strftime('%d-%b-%Y') %></small>
+      <%= link_to 'Transactions', category_path(category), class: "text-text-white ms-3" %>
+    </div>
+  <% end %>
```
Comment (anagudelogu) [inline app/views/categories/index.html.erb]: - [ ] The same happens with your categories view:

|Template|Your design|
|---|---|
| Screenshot from 2023-08-18 21-24-48 | image |

Please consider removing the blank space at the top, the horizontal margin, and moving the button all the way to the bottom of the page! 🤓 🔧

File: app/views/categories/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <li class="list-group-item"><%= expense.name %> - Amount: <%= expense.amount %></li>
+      <% end %>
+    </ul>
+  </div>
+
+
+  
+  <div class="new-trans">
+    <%= link_to 'Add a new expense', new_category_expense_path(@category) %>
+</div>
```
Comment (anagudelogu) [inline app/views/categories/show.html.erb]: - [ ] Please consider doing the same with the button for this view 🤓 🔧 

|Template|Your design|
|---|---|
| Screenshot from 2023-08-18 21-24-48 | image |

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @eudondian, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi, @eudondian 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Requested changes implemented ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**